cmd/list/accessrequests/cmd.go

```diff
@@ -0,0 +1,117 @@
+package accessrequests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	v1 "github.com/openshift-online/ocm-sdk-go/accesstransparency/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+const (
+	use   = "accessrequest"
+	short = "List access requests"
+	long  = "List access requests in Pending or Approved status. " +
+		"If '--cluster' flag is used, list all access requests in any status for the specified cluster."
+	example = `  # List all access requests for cluster 'foo'
+  rosa list accessrequest --cluster foo
+  `
+)
+
+func NewListAccessRequestsCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:     use,
+		Aliases: []string{"access-request", "accessrequests", "access-requests"},
+		Short:   short,
+		Long:    long,
+		Example: example,
+		Run:     rosa.DefaultRunner(rosa.RuntimeWithOCM(), ListAccessRequestsRunner()),
+		Args:    cobra.NoArgs,
+	}
+
+	output.AddFlag(cmd)
+	ocm.AddOptionalClusterFlag(cmd)
+	return cmd
+}
+
+func ListAccessRequestsRunner() rosa.CommandRunner {
+	return func(_ context.Context, r *rosa.Runtime, cmd *cobra.Command, _ []string) error {
+		clusterId := ""
+		if cmd.Flags().Changed("cluster") {
+			cluster, err := r.OCMClient.GetCluster(r.GetClusterKey(), r.Creator)
+			if err != nil {
+				return err
+			}
+			clusterId = cluster.ID()
+		}
+		accessRequests, err := r.OCMClient.ListAccessRequest(clusterId)
+		if err != nil {
+			return err
+		}
+		if output.HasFlag() {
+			output.Print(accessRequests)
+		} else {
+			if len(accessRequests) == 0 {
+				if clusterId == "" {
+					r.Reporter.Infof("There are no Access Requests in Pending or Approved status.")
+				} else {
+					r.Reporter.Infof("There are no Access Requests for cluster '%s'.", r.ClusterKey)
+				}
+				return nil
+			}
+			writer := tabwriter.NewWriter(os.Stdout, 0, 4, 2, ' ', 0)
+			output, hasPending, pendingId := getAccessRequestsOutput(clusterId, accessRequests)
+			fmt.Fprint(writer, output)
+			err = writer.Flush()
+			if err != nil {
+				return err
+			}
+			if hasPending {
+				r.Reporter.Infof("Run the following command to approve or deny the access request:\n\n"+
+					"   rosa create decision --access-request %s --decision Approved\n"+
+					"   rosa create decision --access-request %s --decision Denied\n",
+					pendingId, pendingId)
+			}
+		}
+		return nil
+	}
+}
+
+func getAccessRequestsOutput(clusterId string, accessRequests []*v1.AccessRequest) (string, bool, string) {
```

Comment:
I think we can move this func to the `pkg/ocm/accessrequests.go` file, WDYT?

Reply:
if needed, need to move to place like pkg/accessrequests/output.og. Since this func is not ocm related, pkg/ocm/accessrequests.go is not a good place for it.
Because this func is only for list command and not resuable in other commands, also there are no other utility functions, so I don't create a new folder `accessrequest` under pkg

Reply:
understood- was mainly following convention on the refactors Rob wanted to accomplish, keeping the commands simple and all related funcs in `pkg` (i agree `ocm` is not the best subfolder for this)

But I plan to do a major refactoring of ROSA once we switch teams anyway :smile: